addons/mysterious_egg/static/tests/snippets_menu.test.js

```diff
@@ -0,0 +1,58 @@
+import { setContent } from "@html_editor/../tests/_helpers/selection";
+import { insertText } from "@html_editor/../tests/_helpers/user_actions";
+import { expect, test } from "@odoo/hoot";
+import { animationFrame, click } from "@odoo/hoot-dom";
+import { contains } from "@web/../tests/web_test_helpers";
+import { defineWebsiteModels, openSnippetsMenu, setupWebsiteBuilder } from "./helpers";
+
+defineWebsiteModels();
+
+test("open SnippetsMenu and discard", async () => {
+    await setupWebsiteBuilder(`<h1> Homepage </h1>`);
+    await animationFrame();
+    expect(".o_menu_systray .o-website-btn-custo-primary").toHaveCount(1);
+    await openSnippetsMenu();
+    expect(".o_menu_systray .o-website-btn-custo-primary").toHaveCount(0);
+    await click(".o-snippets-top-actions button:contains(Discard)");
+    await animationFrame();
+    expect(".o_menu_systray .o-website-btn-custo-primary").toHaveCount(1);
+});
+
+test("undo and redo buttons", async () => {
+    const { getEditor } = await setupWebsiteBuilder("<p> Text </p>");
+    await animationFrame();
+    expect(".o_menu_systray .o-website-btn-custo-primary").toHaveCount(1);
+    await openSnippetsMenu();
+    const editor = getEditor();
+    setContent(editor.editable, "<p> Text[] </p>");
+    await insertText(editor, "a");
+    expect(editor.editable).toHaveInnerHTML("<p> Texta </p>");
+    await animationFrame();
+    await click(".o-snippets-menu button.fa-undo");
+    await animationFrame();
+    expect(editor.editable).toHaveInnerHTML("<p> Text </p>");
+    await click(".o-snippets-menu button.fa-repeat");
+    expect(editor.editable).toHaveInnerHTML("<p> Texta </p>");
+});
+
+test("drag and drop snippet inner content", async () => {
+    const { getEditor } = await setupWebsiteBuilder("<div><p>Text</p></div>");
+    await openSnippetsMenu();
+    const editor = getEditor();
```

Comment:
We will probably often have to setup the builder and then open the snippet menu in tests, maybe worth creating a helper for that?